packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1647,9 +1691,11 @@ function bailoutOnAlreadyFinishedWork(
 
   if (current !== null) {
     // Reuse previous context list
-    workInProgress.firstContextDependency = current.firstContextDependency;
+    workInProgress.contextDependencies = current.contextDependencies;
   }
 
+  workInProgress.effectTag &= ~PerformedWork;
```

Comment:
Structuring it this way gets tricky and will make resuming harder. If any pass performed work, we need it to remain as marked as performed work. This risks clearing it if a second pass bails out.

Reply:
Why do you need a per-fiber bit? Can't you just use a global flag?

Reply:
Right, I suppose that’s true. I guess I figured since I had to reset `PerformedWork` in the bailout case anyway I might as well use it for both. But global flag is better.